From 6ad1f2aa8501a2544d78de6dc286ef83ef238245 Mon Sep 17 00:00:00 2001
From: Steve Peters <scpeters@openrobotics.org>
Date: Fri, 12 Apr 2024 14:16:25 -0700
Subject: [PATCH] AudioDecoder: support ffmpeg 7.0

Signed-off-by: Steve Peters <scpeters@openrobotics.org>
---
 gazebo/common/AudioDecoder.cc | 24 +++++++-----------------
 1 file changed, 7 insertions(+), 17 deletions(-)

diff --git a/gazebo/common/AudioDecoder.cc b/gazebo/common/AudioDecoder.cc
index e798d60c27..18ca3b5dd2 100644
--- a/gazebo/common/AudioDecoder.cc
+++ b/gazebo/common/AudioDecoder.cc
@@ -138,12 +138,18 @@ bool AudioDecoder::Decode(uint8_t **_outBuffer, unsigned int *_outBufferSize)
             return false;
         }
 
+#if LIBAVCODEC_VERSION_INT >= AV_VERSION_INT(59, 24, 100)
+        int numChannels = this->codecCtx->ch_layout.nb_channels;
+#else
+        int numChannels = this->codecCtx->channels;
+#endif
+
         // Total size of the data. Some padding can be added to
         // decodedFrame->data[0], which is why we can't use
         // decodedFrame->linesize[0].
         int size = decodedFrame->nb_samples *
           av_get_bytes_per_sample(this->codecCtx->sample_fmt) *
-          this->codecCtx->channels;
+          numChannels;
 
         // Resize the audio buffer as necessary
         if (*_outBufferSize + size > maxBufferSize)
