diff --git a/gazebo/transport/ConnectionManager.cc b/gazebo/transport/ConnectionManager.cc
index 17016a4362..8042c9d6fb 100644
--- a/gazebo/transport/ConnectionManager.cc
+++ b/gazebo/transport/ConnectionManager.cc
@@ -67,6 +67,9 @@ class TopicManagerConnectionTask
   private: msgs::Publish pub;
 };
 
+//////////////////////////////////////////////////
+template class GZ_TRANSPORT_VISIBLE ::SingletonT<gazebo::transport::ConnectionManager>;
+
 //////////////////////////////////////////////////
 ConnectionManager::ConnectionManager()
 {
diff --git a/gazebo/transport/ConnectionManager.hh b/gazebo/transport/ConnectionManager.hh
index 6dc212bc13..45ab732661 100644
--- a/gazebo/transport/ConnectionManager.hh
+++ b/gazebo/transport/ConnectionManager.hh
@@ -34,9 +34,6 @@
 #endif
 #include "gazebo/util/system.hh"
 
-/// \brief Explicit instantiation for typed SingletonT.
-GZ_SINGLETON_DECLARE(GZ_TRANSPORT_VISIBLE, gazebo, transport, ConnectionManager)
-
 namespace gazebo
 {
   namespace transport
@@ -208,4 +205,8 @@ namespace gazebo
     /// \}
   }
 }
+
+/// \brief Explicit instantiation for typed SingletonT.
+extern template class GZ_TRANSPORT_VISIBLE ::SingletonT<gazebo::transport::ConnectionManager>;
+
 #endif
diff --git a/gazebo/transport/TopicManager.cc b/gazebo/transport/TopicManager.cc
index 14fa782fa4..6c036205ee 100644
--- a/gazebo/transport/TopicManager.cc
+++ b/gazebo/transport/TopicManager.cc
@@ -48,6 +48,8 @@ class NodeProcess_TBB
   private: std::vector<NodePtr> *nodes;
 };
 
+//////////////////////////////////////////////////
+template class GZ_TRANSPORT_VISIBLE ::SingletonT<gazebo::transport::TopicManager>;
 
 //////////////////////////////////////////////////
 TopicManager::TopicManager()
diff --git a/gazebo/transport/TopicManager.hh b/gazebo/transport/TopicManager.hh
index b2adc04c8d..c7df09af63 100644
--- a/gazebo/transport/TopicManager.hh
+++ b/gazebo/transport/TopicManager.hh
@@ -44,9 +44,6 @@
 #include "gazebo/transport/Subscriber.hh"
 #include "gazebo/util/system.hh"
 
-/// \brief Explicit instantiation for typed SingletonT.
-GZ_SINGLETON_DECLARE(GZ_TRANSPORT_VISIBLE, gazebo, transport, TopicManager)
-
 namespace gazebo
 {
   namespace transport
@@ -272,4 +269,8 @@ namespace gazebo
     /// \}
   }
 }
+
+/// \brief Explicit instantiation for typed SingletonT.
+extern template class GZ_TRANSPORT_VISIBLE ::SingletonT<gazebo::transport::TopicManager>;
+
 #endif
 
