diff --git a/cmake/SearchForStuff.cmake b/cmake/SearchForStuff.cmake
index 732cbb4e2d..054f0ad6d0 100644
--- a/cmake/SearchForStuff.cmake
+++ b/cmake/SearchForStuff.cmake
@@ -660,11 +660,11 @@ find_package(Qt5 COMPONENTS Core Widgets OpenGL Test REQUIRED)
 ########################################
 # Find Boost, if not specified manually
 include(FindBoost)
-find_package(Boost ${MIN_BOOST_VERSION} REQUIRED thread system filesystem program_options regex iostreams date_time)
+find_package(Boost ${MIN_BOOST_VERSION} REQUIRED thread filesystem program_options regex iostreams date_time)
 
 if (NOT Boost_FOUND)
   set (BUILD_GAZEBO OFF CACHE INTERNAL "Build Gazebo" FORCE)
-  BUILD_ERROR ("Boost not found. Please install thread system filesystem program_options regex iostreams date_time boost version ${MIN_BOOST_VERSION} or higher.")
+  BUILD_ERROR ("Boost not found. Please install thread filesystem program_options regex iostreams date_time boost version ${MIN_BOOST_VERSION} or higher.")
 endif()
 
 ########################################

diff --git a/cmake/gazebo-config.cmake.in b/cmake/gazebo-config.cmake.in
index ac1de2c061..a466287273 100644
--- a/cmake/gazebo-config.cmake.in
+++ b/cmake/gazebo-config.cmake.in
@@ -160,7 +160,7 @@ set(CMAKE_MODULE_PATH
   "${OGRE_INSTALL_PREFIX}/share/OGRE/cmake/modules;${OGRE_INSTALL_PREFIX}/lib/OGRE/cmake;${OGRE_INSTALL_PREFIX}/CMake;${CMAKE_MODULE_PATH}")
 
 # Find boost
-find_package(Boost @MIN_BOOST_VERSION@ REQUIRED thread system
+find_package(Boost @MIN_BOOST_VERSION@ REQUIRED thread
   filesystem program_options regex iostreams date_time)
 list(APPEND @PKG_NAME@_INCLUDE_DIRS ${Boost_INCLUDE_DIRS})
 list(APPEND @PKG_NAME@_LIBRARIES ${Boost_LIBRARIES})
 
