From 5c94221ac3556ae5c90440d60a61913fa54d1335 Mon Sep 17 00:00:00 2001
From: Silvio Traversaro <silvio@traversaro.it>
Date: Sun, 29 Nov 2020 19:03:49 +0100
Subject: [PATCH] Update GazeboUtils.cmake

---
 cmake/GazeboUtils.cmake | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/cmake/GazeboUtils.cmake b/cmake/GazeboUtils.cmake
index 6842ae7c91..f0f66776ba 100644
--- a/cmake/GazeboUtils.cmake
+++ b/cmake/GazeboUtils.cmake
@@ -147,6 +147,10 @@ endmacro ()
 macro (gz_setup_unix)
     # Using dynamic linking in UNIX by default
     set(BUILD_SHARED_LIBS TRUE)
+    # Workaround for https://github.com/conda-forge/cmake-feedstock/issues/129
+    add_link_options("-L${CMAKE_INSTALL_PREFIX}/lib")
+    add_link_options("-Wl,-rpath,${CMAKE_INSTALL_PREFIX}/lib")
+    add_link_options("-Wl,-rpath-link,${CMAKE_INSTALL_PREFIX}/lib")
 endmacro()
 
 #################################################
