diff --git a/cmake/SearchForStuff.cmake b/cmake/SearchForStuff.cmake
index 3d87f4305d..ee4978c4db 100644
--- a/cmake/SearchForStuff.cmake
+++ b/cmake/SearchForStuff.cmake
@@ -71,13 +71,14 @@ else ()
 endif ()
 
 ########################################
-include (FindOpenAL)
-if (NOT OPENAL_FOUND)
-  BUILD_WARNING ("OpenAL not found, audio support will be disabled.")
-  set (HAVE_OPENAL OFF CACHE BOOL "HAVE OpenAL" FORCE)
-else ()
-  set (HAVE_OPENAL ON CACHE BOOL "HAVE OpenAL" FORCE)
-endif ()
+# include (FindOpenAL)
+# if (NOT OPENAL_FOUND)
+#   BUILD_WARNING ("OpenAL not found, audio support will be disabled.")
+#   set (HAVE_OPENAL OFF CACHE BOOL "HAVE OpenAL" FORCE)
+# else ()
+#   set (HAVE_OPENAL ON CACHE BOOL "HAVE OpenAL" FORCE)
+# endif ()
+set (HAVE_OPENAL OFF CACHE BOOL "HAVE OpenAL" FORCE)
 
 ########################################
 find_package(HDF5 COMPONENTS C CXX)
