From e0bec180b612573a053daad73a33bfac860df86e Mon Sep 17 00:00:00 2001
From: Tobias Fischer <info@tobiasfischer.info>
Date: Sun, 2 May 2021 13:55:39 +1000
Subject: [PATCH 1/2] Forward port #2789 part 1

---
 gazebo/common/Material.cc | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/gazebo/common/Material.cc b/gazebo/common/Material.cc
index a32dca1ba0..8ed6d9c383 100644
--- a/gazebo/common/Material.cc
+++ b/gazebo/common/Material.cc
@@ -93,6 +93,10 @@ void Material::SetTextureImage(const std::string &_tex,
       }
     }
   }
+  
+  // normalize the path
+  this->texImage = boost::filesystem::path(this->texImage)
+    .make_preferred().string();
 }
 
 //////////////////////////////////////////////////

From 70dff8dc869a34e78bfbede8784d5b7634ba90e5 Mon Sep 17 00:00:00 2001
From: Tobias Fischer <info@tobiasfischer.info>
Date: Sun, 2 May 2021 13:59:39 +1000
Subject: [PATCH 2/2] Forward port #2789 part 2

---
 gazebo/common/ModelDatabase.cc | 19 ++++++++++++++-----
 1 file changed, 14 insertions(+), 5 deletions(-)

diff --git a/gazebo/common/ModelDatabase.cc b/gazebo/common/ModelDatabase.cc
index 660d156eda..09a0c65efb 100644
--- a/gazebo/common/ModelDatabase.cc
+++ b/gazebo/common/ModelDatabase.cc
@@ -503,19 +503,28 @@ std::string ModelDatabase::GetModelPath(const std::string &_uri,
         continue;
       }
 
+      std::string outputPath = getenv("HOME");
+      outputPath += "/.gazebo/models";
+      
 #ifndef _WIN32
       TAR *tar;
       tar_open(&tar, const_cast<char*>(tarfilename.c_str()),
           nullptr, O_RDONLY, 0644, TAR_GNU);
 
-      std::string outputPath = getenv("HOME");
-      outputPath += "/.gazebo/models";
-
       tar_extract_all(tar, const_cast<char*>(outputPath.c_str()));
+#else
+      // Tar now is a built-in tool since Windows 10 build 17063.
+      std::string cmdline = "tar xzf \"";
+      cmdline += tarfilename + "\" -C \"";
+      cmdline += outputPath + "\"";
+      auto ret = system(cmdline.c_str());
+      if (ret != 0)
+      {
+        gzerr << "tar extract ret = " << ret << ", cmdline = " << cmdline << std::endl;
+      }
+#endif
       path = outputPath + "/" + modelName;
-
       ModelDatabase::DownloadDependencies(path);
-#endif
     }
 
     curl_easy_cleanup(curl);
