diff --git a/media/materials/scripts/shadow_caster_ignore_heightmap.program b/media/materials/scripts/shadow_caster_ignore_heightmap.program
index 26ab9058ca..6fe04097c8 100644
--- a/media/materials/scripts/shadow_caster_ignore_heightmap.program
+++ b/media/materials/scripts/shadow_caster_ignore_heightmap.program
@@ -1,14 +1,3 @@
-vertex_program shadow_caster_vp_glsl glsl
-{
-  source shadow_caster_vp.glsl
-
-  default_params
-  {
-    param_named_auto world_view_proj_mat worldviewproj_matrix
-    param_named_auto texel_offsets texel_offsets
-  }
-}
-
 fragment_program shadow_caster_ignore_heightmap_fp_glsl glsl
 {
   source shadow_caster_ignore_heightmap_fp.glsl
