diff --git a/cmake/SearchForStuff.cmake b/cmake/SearchForStuff.cmake
index 2b79fd4be3..d55e8ce8e3 100644
--- a/cmake/SearchForStuff.cmake
+++ b/cmake/SearchForStuff.cmake
@@ -541,14 +542,15 @@ if (PKG_CONFIG_FOUND)
 
   ########################################
   # Find GNU Triangulation Surface Library
-  pkg_check_modules(gts gts)
-  if (gts_FOUND)
-    message (STATUS "Looking for GTS - found")
-    set (HAVE_GTS TRUE)
-  else ()
-    set (HAVE_GTS FALSE)
-    BUILD_WARNING ("GNU Triangulation Surface library not found - Gazebo will not have CSG support.")
-  endif ()
+  # pkg_check_modules(gts gts)
+  # if (gts_FOUND)
+  #   message (STATUS "Looking for GTS - found")
+  #   set (HAVE_GTS TRUE)
+  # else ()
+  #   set (HAVE_GTS FALSE)
+  #   BUILD_WARNING ("GNU Triangulation Surface library not found - Gazebo will not have CSG support.")
+  # endif ()
+  set (HAVE_GTS FALSE)
 
   #################################################
   # Find bullet
@@ -669,15 +671,16 @@ add_manpage_target()
 
 ########################################
 # Find Space Navigator header and library
-find_library(SPNAV_LIBRARY NAMES spnav)
-find_file(SPNAV_HEADER NAMES spnav.h)
-if (SPNAV_LIBRARY AND SPNAV_HEADER)
-  message(STATUS "Looking for libspnav and spnav.h - found")
-  set(HAVE_SPNAV TRUE)
-else()
-  message(STATUS "Looking for libspnav and spnav.h - not found")
-  set(HAVE_SPNAV FALSE)
-endif()
+# find_library(SPNAV_LIBRARY NAMES spnav)
+# find_file(SPNAV_HEADER NAMES spnav.h)
+# if (SPNAV_LIBRARY AND SPNAV_HEADER)
+#   message(STATUS "Looking for libspnav and spnav.h - found")
+#   set(HAVE_SPNAV TRUE)
+# else()
+#   message(STATUS "Looking for libspnav and spnav.h - not found")
+#   set(HAVE_SPNAV FALSE)
+# endif()
+set(HAVE_SPNAV FALSE)
 
 ########################################
 # Find xsltproc, which is used by tools/check_test_ran.py
diff --git a/cmake/gazebo-config.cmake.in b/cmake/gazebo-config.cmake.in
index 4fa403cfde..b571840557 100644
--- a/cmake/gazebo-config.cmake.in
+++ b/cmake/gazebo-config.cmake.in
@@ -129,7 +129,7 @@ endif()
 
 # Visual Studio enables c++11 support by default
 if (NOT MSVC)
-  list(APPEND @PKG_NAME@_CXX_FLAGS -std=c++11)
+  list(APPEND @PKG_NAME@_CXX_FLAGS)
 endif()
 if ("${CMAKE_CXX_COMPILER_ID}" MATCHES "Clang" AND
     "${CMAKE_SYSTEM_NAME}" MATCHES "Darwin")
 
