From 183122dc0d6f96d030c48b1a09d3d46d962fad45 Mon Sep 17 00:00:00 2001
From: Silvio Traversaro <silvio.traversaro@iit.it>
Date: Thu, 26 Sep 2019 20:49:28 +0000
Subject: [PATCH] Add support for finding TBB via CMake config files

---
 cmake/SearchForStuff.cmake | 24 +++++++++++++++++-------
 1 file changed, 17 insertions(+), 7 deletions(-)

diff --git a/cmake/SearchForStuff.cmake b/cmake/SearchForStuff.cmake
index edc5a98dff..3ab3682f95 100644
--- a/cmake/SearchForStuff.cmake
+++ b/cmake/SearchForStuff.cmake
@@ -312,13 +312,23 @@ if (PKG_CONFIG_FOUND)
     message(STATUS "TBB not found, attempting to detect manually")
     set (TBB_PKG_CONFIG "")
 
-    find_library(tbb_library tbb ENV LD_LIBRARY_PATH)
-    if (tbb_library)
-      set(TBB_FOUND true)
-      set(TBB_LIBRARIES ${tbb_library})
-    else (tbb_library)
-      BUILD_ERROR ("Missing: TBB - Threading Building Blocks")
-    endif(tbb_library)
+    # Workaround for CMake bug https://gitlab.kitware.com/cmake/cmake/issues/17135
+    unset(TBB_FOUND CACHE)
+
+    find_package(TBB CONFIG)
+    if (TBB_FOUND)
+      set(TBB_LIBRARIES TBB::tbb)
+    endif()
+
+    if (NOT TBB_FOUND)
+      find_library(tbb_library tbb ENV LD_LIBRARY_PATH)
+      if (tbb_library)
+        set(TBB_FOUND true)
+        set(TBB_LIBRARIES ${tbb_library})
+      else (tbb_library)
+        BUILD_ERROR ("Missing: TBB - Threading Building Blocks")
+      endif(tbb_library)
+    endif (NOT TBB_FOUND)
   endif (NOT TBB_FOUND)
 
   #################################################
