From 075c13de5c5553f6290c1a53a39d9165fc8cfd76 Mon Sep 17 00:00:00 2001
From: Silvio Traversaro <silvio@traversaro.it>
Date: Tue, 2 Mar 2021 19:13:27 +0100
Subject: [PATCH] Fix missing link to OpenAL in gazebo_util library

---
 gazebo/util/CMakeLists.txt | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/gazebo/util/CMakeLists.txt b/gazebo/util/CMakeLists.txt
index 2c69f2b250..5831fc1aab 100644
--- a/gazebo/util/CMakeLists.txt
+++ b/gazebo/util/CMakeLists.txt
@@ -85,6 +85,10 @@ target_link_libraries(gazebo_util
   ${IGNITION-MSGS_LIBRARIES}
 )
 
+if (HAVE_OPENAL)
+  target_link_libraries(gazebo_util ${OPENAL_LIBRARY})
+endif()
+
 # define if tinxml2 major version >= 6
 # https://github.com/ignitionrobotics/ign-common/issues/28
 if (NOT tinyxml2_VERSION VERSION_LESS "6.0.0")
 
