--- cmake/GazeboUtils.cmake
+++ cmake/GazeboUtils.cmake
@@ -115,7 +115,11 @@
 #################################################
 macro (gz_install_library _name)
   set_target_properties(${_name} PROPERTIES SOVERSION ${GAZEBO_MAJOR_VERSION} VERSION ${GAZEBO_VERSION_FULL})
-  install (TARGETS ${_name} DESTINATION ${LIB_INSTALL_DIR} COMPONENT shlib)
+  install (TARGETS ${_name}
+    ARCHIVE DESTINATION ${LIB_INSTALL_DIR}
+    LIBRARY DESTINATION ${LIB_INSTALL_DIR}
+    RUNTIME DESTINATION ${BIN_INSTALL_DIR}
+    COMPONENT shlib)
 endmacro ()
 
 #################################################
@@ -142,11 +146,9 @@
     # Don't pull in the Windows min/max macros
     add_definitions(-DNOMINMAX)
 
-    #use static libraries for FREEIMAGE
-    add_definitions(-DFREEIMAGE_LIB)
-
     # Use dynamic linking for boost
     add_definitions(-DBOOST_ALL_DYN_LINK)
+    add_definitions(-DBOOST_IOSTREAMS_NO_LIB=1)
 
     # Use dynamic linking for protobuf
     add_definitions(-DPROTOBUF_USE_DLLS)
